### PR TITLE
Add instance id label to osg-hosted-ce

### PR DESCRIPTION
**Pull request by jmcragun**

### CODE REVIEW

File: stable/osg-hosted-ce/osg-hosted-ce/templates/deployment.yaml [hunk 1]
```diff
@@ -173,13 +170,11 @@ spec:
...
           mountPath: /var/log/condor-ce
         {{ end }}
-        {{ if .Values.BoscoOverrides.Enabled }}
-        {{ if .Values.BoscoOverrides.GitKeySecret }}
+        {{ if .Values.BoscoOverrides.RepoNeedsPrivKey }}
         - name: osg-hosted-ce-{{ .Values.Instance }}-gitkey
           mountPath: /etc/osg/git.key
           subPath: git.key
         {{ end }}
-        {{ end }}
```
Comment (brianhlin) [inline stable/osg-hosted-ce/osg-hosted-ce/templates/deployment.yaml:182]: This appears to partially revert https://github.com/slateci/slate-catalog/commit/d367601857d8e7461f3c89c29c92cf02b790dc90#diff-3bb7f07a3f0bc43fe03b034bdcc8a765bbcb1fdb65a2afcaf0fbc61d53c49176. Is that intended?

File: stable/osg-frontier-squid/osg-frontier-squid/templates/deployment.yaml [hunk 2]
```diff
@@ -37,7 +38,7 @@ spec:
       containers:
       # Container for the primary application, OSG Frontier Squid
       - name: osg-frontier-squid
-        image: opensciencegrid/frontier-squid:20210106-2231
+        image: opensciencegrid/frontier-squid:fresh
```
Comment (brianhlin) [inline stable/osg-frontier-squid/osg-frontier-squid/templates/deployment.yaml:41]: I know @LincolnBryant had some issues with the certain tags of `frontier-squid`, do we actually want to make this change? It also seems unrelated to adding an instance ID label

File: stable/osg-hosted-ce/osg-hosted-ce/templates/deployment.yaml [hunk 3]
```diff
@@ -23,10 +24,6 @@ spec:
         release: {{ .Release.Name }}
         instance: {{ .Values.Instance }}
     spec:
-      {{ if .Values.NodeSelection }}
-      nodeSelector: 
-        kubernetes.io/hostname: {{ .Values.NodeSelection }}
-      {{ end }}
```
Comment (brianhlin) [inline stable/osg-hosted-ce/osg-hosted-ce/templates/deployment.yaml:29]: Do we want to remove this capability? If so, I think it belongs in a separate PR or commit

File: stable/osg-hosted-ce/osg-hosted-ce/values.yaml [hunk 4]
```diff
@@ -160,3 +156,9 @@ ContainerTags:
 # - Generates a test CA and self-signed host cert/key pair
 Developer:
   Enabled: false
+
+### SLATE-START ###
+SLATE:
+  Instance:
+    ID: "untagged"
+### SLATE-END ###
```
Comment (brianhlin) [inline stable/osg-hosted-ce/osg-hosted-ce/values.yaml:164]: Could you add some comments to indicate the purpose of this configuration? What do the `SLATE-START` and `SLATE-END` stanzas indicate?

**Review by brianhlin — CHANGES_REQUESTED** [review]
I've got some questions about what the instance ID is for and it looks like there are a few other changes that snuck in

File: stable/osg-hosted-ce/osg-hosted-ce/values.yaml (hunk 4 above)
Comment (Mansalu) [reply stable/osg-hosted-ce/osg-hosted-ce/values.yaml:164]: Basically a "do-not-modify" section. The SLATE API itself looks at this part of the Values, and can inject values at run-time. In particular, we want to have SLATE tag each object it creates in Kubernetes with the unique instance ID it belongs to.

File: stable/osg-hosted-ce/osg-hosted-ce/templates/deployment.yaml (hunk 3 above)
Comment (Mansalu) [reply stable/osg-hosted-ce/osg-hosted-ce/templates/deployment.yaml:29]: I think we want to keep this.

File: stable/osg-frontier-squid/osg-frontier-squid/Chart.yaml [hunk 5]
```diff
@@ -5,4 +5,4 @@ appVersion: 4.4-2.1
 description: A Helm chart for configuration and deployment of the Open Science Grid's Frontier Squid application.
 name: osg-frontier-squid
 # Chart version
-version: 1.4.6
+version: 1.4.2
```
Comment (Mansalu) [inline stable/osg-frontier-squid/osg-frontier-squid/Chart.yaml:8]: Versions should *never* decrement.

**Review by Mansalu — CHANGES_REQUESTED** [review]
Looks like this is rebasing against and old version of the chart. This reverts a lot of things we want to keep and actually decrements several versions.

File: stable/osg-hosted-ce/osg-hosted-ce/values.yaml (hunk 4 above)
Comment (brianhlin) [reply stable/osg-hosted-ce/osg-hosted-ce/values.yaml:164]: Ah ok, that's what I suspected. It'd be nice if you guys could add a layer of indirection to inject these values behind the scenes so that they don't have to be exposed to the end users

Comment (Mansalu) [reply stable/osg-hosted-ce/osg-hosted-ce/values.yaml:164]: I think the values need to appear in the chart for the base chart to work on vanilla Helm.

**Comment by Mansalu** [comment]
I also noticed this is both frontier squid and hosted-ce in the same patch.

**Comment by jmcragun** [comment]
Yeah, this PR got a bit messed up and at this point it's probably easier for me to just start fresh. This PR wasn't mean to invoke so many changes (the reason it did was that I forgot to pull master before pushing these changes). Sorry everyone!
